x-pack/plugins/cases/server/client/factory.ts

```diff
@@ -160,4 +159,62 @@ export class CasesClientFactory {
       attachmentService,
     };
   }
+
+  /**
+   * This function attempts to retrieve the current user's info. The first method is using the user profile api
+   * provided by the security plugin. If that fails or the session isn't found then we will attempt using authc
+   * which will not retrieve the profile uid but at least gets us the username and sometimes full name, and email.
+   *
+   * This function also forces the fields to be strings or null (except the profile uid since it's optional anyway)
+   * because the get case API expects a created_by field to be set. If we leave the fields as undefined
+   * then the resulting object in ES will just be empty and it'll fail to encode the user when returning it to the API
+   * request. If we force them to be null it will succeed.
+   */
+  private async getUserInfo(request: KibanaRequest): Promise<{
+    username: string | null;
+    full_name: string | null;
+    email: string | null;
+    profile_uid?: string;
+  }> {
+    if (!this.isInitialized || !this.options) {
+      throw new Error('CasesClientFactory must be initialized before calling create');
+    }
```

Comment:
It seems that we are using this check a lot in the factory class. Maybe we can extract it to a function.